diff --git a/gentype.exe b/gentype.exe
deleted file mode 100755
index 30df9c7966c5cab032df6235c5d575029cac2a43..0000000000000000000000000000000000000000
--- a/gentype.exe
+++ /dev/null
@@ -1,2 +0,0 @@
-echo "gentype was not installed correctly"
-exit 1
diff --git a/postinstall.js b/postinstall.js
index 2c29d18ffd10c9d356ef98ac976efe6321fc5c06..a5451170bb79053de4ba0d0a68dcb5b98b29d445 100644
--- a/postinstall.js
+++ b/postinstall.js
@@ -1,35 +1,37 @@
-var path = require("path");
-var fs = require("fs");
+var path = require('path');
+var fs = require('fs');
 var platform = process.platform;
 
 /* We always use the .exe extension, no matter if unix / win32 */
-const targetPath = path.join(__dirname, "gentype.exe");
+const targetPath = path.join(__dirname, 'gentype.exe');
 
 function fail(msg) {
-    console.warn(msg);
-    process.exit(1);
+  console.warn(msg);
+  process.exit(1);
 }
 
 function getPlatformBinaryPath(platform) {
-  return path.join(__dirname, "vendor-" + platform, "gentype.exe");
+  return path.join(__dirname, 'vendor-' + platform, 'gentype.exe');
 }
 
 function movePlatformBinary(platform) {
   const sourcePath = getPlatformBinaryPath(platform);
 
-  if(!fs.existsSync(sourcePath)) {
-      return fail("error: executable not found: " + sourcePath);
+  if (!fs.existsSync(sourcePath)) {
+    // I have to rebuild rescript manually...
+    console.warn('error: executable not found: ' + sourcePath);
+    return;
   }
   fs.renameSync(sourcePath, targetPath);
   fs.chmodSync(targetPath, 0777);
 }
 
 switch (platform) {
-  case "win32":
-  case "linux":
-  case "darwin":
+  case 'win32':
+  case 'linux':
+  case 'darwin':
     movePlatformBinary(platform);
     break;
   default:
-    fail("error: no release built for the " + platform + " platform");
+    fail('error: no release built for the ' + platform + ' platform');
 }
